### ISSUE TITLE
Cannot add year to birthdate without a year

### ISSUE THREAD

**Issue by caugner** [nextcloud/contacts#1041]
### Describe the bug

**To Reproduce**
1. Create a contact with a birthdate *without* a year (using CardDAV).
2. Open the contact in Nextcloud.
3. Edit the birthdate, *add* the year and save.
4. The birthdate doesn't change.

**Expected behavior**
The changed date *with* the year should be saved.

**Actual behavior**
The date does not change.

**Screenshots**
Not applicable.

### Server configuration

**Operating system**: Linux

**Web server:** Apache

**Database:** MySQL

**PHP version:** 7.2

**Nextcloud version:** 15.0.6

**Contacts version:**  3.0.5

**Updated from an older Nextcloud or fresh install:** Updated from 15.0.5

**Signing status:**
```
No errors have been found.
```

**List of activated apps:**
```
Enabled:
  - accessibility: 1.1.0
  - activity: 2.8.2
  - admin_audit: 1.5.0
  - calendar: 1.6.4
  - cloud_federation_api: 0.1.0
  - comments: 1.5.0
  - contacts: 3.0.5
  - dav: 1.8.1
  - federatedfilesharing: 1.5.0
  - federation: 1.5.0
  - files: 1.10.0
  - files_pdfviewer: 1.4.0
  - files_sharing: 1.7.0
  - files_texteditor: 2.7.0
  - files_trashbin: 1.5.0
  - files_versions: 1.8.0
  - files_videoplayer: 1.4.0
  - firstrunwizard: 2.4.0
  - gallery: 18.2.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.3.0
  - nextcloud_announcements: 1.4.0
  - notes: 2.5.1
  - notifications: 2.3.0
  - oauth2: 1.3.0
  - password_policy: 1.5.0
  - provisioning_api: 1.5.0
  - serverinfo: 1.5.0
  - sharebymail: 1.5.0
  - support: 1.0.0
  - survey_client: 1.3.0
  - systemtags: 1.5.0
  - theming: 1.6.0
  - twofactor_backupcodes: 1.4.1
  - updatenotification: 1.5.0
  - workflowengine: 1.5.0
Disabled:
  - bruteforcesettings
  - encryption
  - files_external
  - tasks
  - user_ldap
```

**Nextcloud configuration:**
```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "overwritehost": "***REMOVED SENSITIVE VALUE***",
        "overwriteprotocol": "https",
        "overwritewebroot": "",
        "overwritecondaddr": "",
        "forcessl": "true",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "15.0.6.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "loglevel": 2,
        "theme": "",
        "maintenance": false,
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "forceSSLforSubdomains": true,
        "trashbin_retention_obligation": "auto",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "updater.release.channel": "stable",
        "mail_smtpsecure": "tls",
        "mail_sendmailmode": "smtp",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "PLAIN",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}
```

### Client configuration
**Browser:** Firefox 66.0.2

**Operating system:** Linux

**CardDAV-clients:** DAVx⁵ 2.4.0.1-gplay (280)

#### *Web server error log*
(omitted, no errors)

#### *data/nextcloud.log*
(omitted, no errors)

#### Browser log
(omitted, no errors)

**Comment by skjnldsv** [comment]
Hey! :)
Thanks for your feedback!

Are we talking about those kind of props: `BDAY;X-APPLE-OMIT-YEAR=1604:1604-06-20` or `BDAY:--0620` ?

**Comment by caugner** [comment]
@skjnldsv I have just checked and it's `BDAY:--0620`.

**Comment by skjnldsv** [comment]
will be fixed alongside with #625
